notebooks/scrf.ipynb

```diff
@@ -0,0 +1,828 @@
+    "    print(\"reprojecting\")\n",
+    "    field = field.transpose(\"time\", \"y\", \"x\").rio.reproject(\"EPSG:4326\")\n",
+    "    field = field.rename({\"x\": \"lon\", \"y\": \"lat\"})\n",
+    "    template = template.rename({\"x\": \"lon\", \"y\": \"lat\"})\n",
+    "\n",
+    "    # regridding\n",
+    "    print(\"regridding\")\n",
+    "    regridder = xe.Regridder(\n",
+    "        field, template, \"bilinear\", extrap_method=\"nearest_s2d\"\n",
+    "    )\n",
+    "    regridded_field = regridder(field)\n",
+    "\n",
+    "    print(\"saving\")\n",
+    "    # save to file\n",
+    "    if variable == \"temp\":\n",
+    "        for v in [\"tasmax\", \"tasmin\"]:\n",
+    "            path = make_scrf_path(\"ERA5\", v, start_year, end_year)\n",
+    "            store = fsspec.get_mapper(\"az://flow-outputs/intermediate/\" + path)\n",
+    "            regridded_field.to_dataset(name=\"scrf\").to_zarr(\n",
+    "                store, mode=\"w\", consolidated=True\n",
+    "            )\n",
+    "    else:\n",
+    "        path = make_scrf_path(\"ERA5\", \"pr\", start_year, end_year)\n",
+    "        store = fsspec.get_mapper(\"az://flow-outputs/intermediate/\" + path)\n",
+    "        regridded_field.to_dataset(name=\"scrf\").to_zarr(\n",
+    "            store, mode=\"w\", consolidated=True\n",
+    "        )\n",
+    "\n",
+    "    return (\"processed\", path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9e61ad51-deb2-4e76-9130-8ccf62cf3586",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "for variable in [\"temp\", \"pr\"]:\n",
+    "    for start_year in np.arange(1981, 2100, 10):\n",
+    "        end_year = str(start_year + 9)\n",
+    "        start_year = str(start_year)\n",
+    "        print(variable, start_year, end_year)\n",
+    "\n",
+    "        generate_scrf(\n",
+    "            variable=variable,\n",
+    "            spatial_scales=spatial_scales,\n",
+    "            temporal_scales=temporal_scales,\n",
+    "            start_year=start_year,\n",
+    "            end_year=end_year,\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cebadadf-4a57-48d4-96fc-c7344135614b",
+   "metadata": {},
+   "source": [
+    "# Validating correlation length\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e1f48345-5c73-414e-a13b-311049b736cc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get the random field output\n",
+    "# make sure the different decades are different\n",
+    "# make sure precip is different from tmax, but tmax is the same as tmin\n",
+    "time = 100\n",
+    "\n",
+    "ds1 = xr.open_zarr(\n",
+    "    \"az://flow-outputs/intermediate/scrf/ERA5_pr_2011_2020.zarr\"\n",
+    ").scrf.isel(time=time)\n",
+    "\n",
+    "ds2 = xr.open_zarr(\n",
+    "    \"az://flow-outputs/intermediate/scrf/ERA5_pr_2021_2030.zarr\"\n",
```

Comment:
if we wanted to do the full analysis through 2100 we would extend this in decadal increments?

Reply:
i generated data from 1850-2100